src/xb-builder-node.h

```diff
@@ -51,15 +38,50 @@ typedef enum {
 	XB_BUILDER_NODE_FLAG_LAST
 } XbBuilderNodeFlags;
 
+// XbBuilderNodes are deliberately not part of the GObject system. They are plain C structs
+// this is because there are potentially tens of thousands of them and the overhead of GObject
+// per-object RAM overhead and memory management becomes significant at that scale.
+// In versions 0.3.22 and earler these were GObjects, and profiling showed this was the main
+// bottleneck to parser performance.
```

Comment:
This comment should probably become part of a gtk-doc comment for `XbBuilderNode`, in the C file.